cast/tests/e2e/helper_functions.rs

```diff
@@ -0,0 +1,128 @@
+use crate::helpers::devnet::URL;
```

Comment:
Sorry but I do not understand this file, it is very misleading. You called it `helper_functions.rs` but all I see here are tests. It doesn't make sense
Could you explain why did you move all these tests to a separate file in the first place? It is basically against the rust convention that says we should have such tests in the same file as functions. Can we move them back there?

Reply:
Also, tests here don't look like e2e tests, are you sure they are e2e?

Reply:
Sure, happy to help.

I have named this file helper_functions, as it is supposed to test all the helper functions we have (and I was not sure I can name it lib to idjicate those helper functions are from lib exclusively). This resides in test directory, where all the test files are present. If you have a better idea how to name it, let me know :) 

>Could you explain why did you move all these tests to a separate file in the first place? It is basically against the rust convention that says we should have such tests in the same file as functions. Can we move them back there?

the convention is that all the unit tests for specific file are to be in this file, but e2e tests should not be there + I nned them to be in e2e directory since 1. they are e2e 2. They require devnet, which is assured only for tests in e2e directory. We cannot, and we should not move them back, becasue they will break.

> 
Also, tests here don't look like e2e tests, are you sure they are e2e?

they are, they all setup up a provider, which makes them e2e.

Reply:
Discussed offline, and agreed we should move this to the `integration` folder as they are indeed integration tests, not e2e.

> If you have a better idea how to name it, let me know :)

Sure, I think this should be named `cast/tests/integration/lib_tests.rs` because it tests functions from `lib.rs`. `helper_functions.rs` doesn't tell much tbh.

Reply:
> the convention is that all the unit tests for specific file are to be in this file, but e2e tests should not be there

Yes, this is correct, thanks for clarifying, that also applies to integration tests.